samples/word/10-content-controls/insert-and-change-checkbox-content-control.yaml

```
@@ -0,0 +1,246 @@
+              })
+              .getFirstOrNullObject();
+            selectedContentControl.load("id");
+
+            await context.sync();
+
+            if (selectedContentControl.isNullObject) {
+              const parentContentControl = selectedRange.parentContentControl;
+              parentContentControl.load("id,type");
+              await context.sync();
+
+              if (parentContentControl.isNullObject || parentContentControl.type !== Word.ContentControlType.checkBox) {
+                console.warn("No checkbox content control is currently selected.");
+                return;
+              } else {
+                selectedContentControl = parentContentControl;
+              }
+            }
+
+            console.log(`About to delete checkbox content control with id: ${selectedContentControl.id}`);
+            selectedContentControl.delete(false);
+            await context.sync();
+
+            console.log("Deleted checkbox content control.");
+          });
+        }
+
+        async function setup() {
+          await Word.run(async (context) => {
+            const body = context.document.body;
+            body.clear();
+            body.insertParagraph("Task 3", "Start");
+            body.insertParagraph("Task 2", "Start");
+            body.insertParagraph("Task 1", "Start");
+            body.paragraphs.getLast().insertText("Task 4", "Replace");
+          });
+        }
+
+        // Default helper for invoking an action and handling errors.
+        async function tryCatch(callback) {
+          try {
+            await callback();
+          } catch (error) {
+            // Note: In a production add-in, you'd want to notify the user through your add-in's UI.
+            console.error(error);
+          }
+        }
+    language: typescript
+template:
+    content: |-
+        <section class="ms-font-m">
+            This sample demonstrates how to insert, change, and delete checkbox content controls.
+        </section>
+
+        <section class="setup ms-font-m">
+            <h3>Set up</h3>
+            <button id="setup" class="ms-Button">
+              <span class="ms-Button-label">Setup</span>
+          </button>
+        </section>
+
+        <section class="samples ms-font-m">
+            <h3>Try it out</h3>
+            <p>
+                <span class="ms-font-m">Insert checkbox content controls on each paragraph.</span>
+                <button id="insert-controls" class="ms-Button">
+            <span class="ms-Button-label">Insert</span>
+          </button>
+            </p>
+            <p>
+                <div>1. Select content that includes at least one checkbox content control.</div>
+                <span class="ms-font-m">2. Modify isChecked property of the first checkbox content control found in the selection.</span>
+                <button id="toggle-control" class="ms-Button">
+                        <span class=ms-Button-label>Toggle</span>
+                    </button>
```

I tend to avoid instructions in the task pane. I think you can have the same effect by adding a bit more to the button text (see suggestion). As for the clarity around "first versus all selected", I think the code makes the behavior clear.

```suggestion
                <button id="toggle-control" class="ms-Button">
                        <span class=ms-Button-label>Toggle selected checkbox</span>
                    </button>
```